### PR TITLE
Fix working with LDAP replica server

### PR DESCRIPTION
**Pull request by jvillafanez**
Manual port of https://github.com/owncloud/core/pull/27598

Original issue: https://github.com/owncloud/core/issues/27575

Not tested yet with 10.0 (the original patch was partially tested with 9.1 -> https://github.com/owncloud/core/pull/27598#issuecomment-292905373)

### CODE REVIEW

**Comment by butonic** [comment]
ok, works, but the `bind()` method was binding twice. I simplified the connection establishment further to fix that.

sync & login with multiple servers works.

File: lib/Connection.php [hunk 1]
```diff
@@ -520,42 +523,51 @@ private function establishConnection() {
 				}
 			}
 
-			$bindStatus = false;
-			$error = -1;
```
Comment (butonic) [inline lib/Connection.php]: the whole connect logic can be done without the tracking of bind status er error codes ...

File: lib/Connection.php [hunk 2]
```diff
@@ -580,37 +592,26 @@ private function doConnect($host, $port) {
...
-			return false;
-		}
-		$getConnectionResourceAttempt = true;
+
 		$cr = $this->getConnectionResource();
-		$getConnectionResourceAttempt = false;
 		if(!$this->getLDAP()->isResource($cr)) {
 			return false;
 		}
-		$ldapLogin = @$this->getLDAP()->bind($cr,
```
Comment (jvillafanez) [inline lib/Connection.php]: Bind function not doing `bind()`? This needs a comment: binding is done via `getConnectionResource()` (if I'm not mistaken)

**Comment by jvillafanez** [comment]
No further complains as long as it works.

Note that we might need to backport the commit to stable9.1 for the sake fo consistency.

**Comment by tomneedham** [comment]
rebased, tested with replica docker server locally, killed the original container and with this PR everything goes back to normal :)

**Pull request merged**